docker/builder/builder.go

```diff
@@ -52,9 +54,30 @@ func (d *DaemonBuilder) Build(ctx context.Context, imageName string) error {
 		return err
 	}
 	defer buildCtx.Close()
+	var l logger.Logger
+	if d.LoggerFactory != nil {
+		l = d.LoggerFactory.CreateBuildLogger(imageName)
+	}
 
 	var progBuff io.Writer = os.Stdout
```

Comment:
Hmmm, we could try and enforce a logger factory. Maybe if we move some components to constructors where we can verify the fields?